File: delivery-service/delivery/db/src/main/java/org/example/delivery/db/userordermenu/UserOrderMenuRepository.java
package org.example.delivery.db.userordermenu;

import org.example.delivery.db.userordermenu.enums.UserOrderMenuStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface UserOrderMenuRepository extends JpaRepository<UserOrderMenuEntity, Long> {

    /**
     * SELECT * FROM user_order_menu WHERE user_order_id = ? AND status = ?;
     */
    List<UserOrderMenuEntity> findAllByUserOrderIdAndStatus(Long userOrderId, UserOrderMenuStatus status);
}

File: delivery/db/src/main/kotlin/org/example/delivery/db/userordermenu/UserOrderMenuRepository.kt
package org.example.delivery.db.userordermenu

import org.example.delivery.db.userordermenu.enums.UserOrderMenuStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserOrderMenuRepository : JpaRepository<UserOrderMenuEntity, Long> {
    /**
     * SELECT * FROM user_order_menu WHERE user_order_id = ? AND status = ?;
     */
    fun findAllByUserOrderIdAndStatus(userOrderId: Long?, status: UserOrderMenuStatus?): List<UserOrderMenuEntity>
}
